src/main/java/logic/MainLogic.java

```diff
@@ -147,7 +147,8 @@ private static boolean initializeSystem (String token) {
             System.out.println("Error in received initialization token");
         }
         if (inputString.isBlank() || inputString.isEmpty()) {
-            throw new IllegalArgumentException("Input cannot be empty!\n");
+            System.out.println("Input cannot be empty!");
+            return false;
```

Comment:
Ok, so instead of throwing an exception you want to output to the terminal, could consider adding to a logger too!

Reply:
Good idea, will add a logger. I output to the terminal instead of throwing an exception because I feel it's easier to prompt the user for the input again.